packages/react/src/components/FileUploader/FileUploader.js

```diff
@@ -192,72 +193,58 @@ export class FileUploaderButton extends Component {
   }
 }
 
-export class Filename extends Component {
-  static propTypes = {
-    /**
-     * Specify an optional object of styles to be applied inline to the root
-     * node
-     */
-    style: PropTypes.object,
-
-    /**
-     * Specify the status of the File Upload
-     */
-    status: PropTypes.oneOf(['edit', 'complete', 'uploading']),
-
-    /**
-     * Provide a description for the complete/close icon that can be read by screen readers
-     */
-    iconDescription: PropTypes.string,
-  };
-
-  static defaultProps = {
-    onKeyDown: () => {},
-    status: 'uploading',
-    style: {},
-    tabIndex: 0,
-  };
-
-  render() {
-    const { iconDescription, status, style, ...other } = this.props;
-
-    if (status === 'uploading') {
+export function Filename({ iconDescription, status, ...other }) {
+  switch (status) {
+    case 'uploading':
+      return <InlineLoading iconDescription={iconDescription} />;
+    case 'edit':
       return (
-        <div
-          className={`${prefix}--loading`}
-          style={{ ...style, width: '1rem', height: '1rem' }}
-          {...other}>
-          <svg className={`${prefix}--loading__svg`} viewBox="-42 -42 84 84">
-            <circle cx="0" cy="0" r="37.5" />
-          </svg>
-        </div>
-      );
-    } else if (status === 'edit') {
-      return (
-        <CloseFilled16
+        <Close16
           className={`${prefix}--file-close`}
           aria-label={iconDescription}
-          style={style}
           {...other}>
           {iconDescription && <title>{iconDescription}</title>}
-        </CloseFilled16>
+        </Close16>
       );
-    } else if (status === 'complete') {
+    case 'complete':
       return (
         <CheckmarkFilled16
           className={`${prefix}--file-complete`}
           aria-label={iconDescription}
-          style={style}
           {...other}>
           {iconDescription && <title>{iconDescription}</title>}
         </CheckmarkFilled16>
       );
-    } else {
+    default:
       return null;
-    }
   }
 }
 
+Filename.propTypes = {
+  /**
+   * Provide a description of the SVG icon to denote file upload status
+   */
+  iconDescription: PropTypes.string,
+
+  /**
+   * Provide an optional `onKeyDown` hook that is called if Space or Return is
+   * pressed while the component is focused
+   */
+  status: PropTypes.oneOf(['edit', 'complete', 'uploading']),
+
+  /**
+   * Provide a custom tabIndex value for the <Filename>
+   */
+  tabIndex: PropTypes.string,
+};
+
+Filename.defaultProps = {
+  iconDescription: 'Uploading file',
+  onKeyDown: () => {},
```

Comment:
Should we just check if `onKeyDown` is not undefined and conditionally call it?

Reply:
this was carried over from `FileUploaderItem` but I just removed it